data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/PlainTextConsumer.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.consumer;
+import org.apache.kafka.clients.consumer.CommitFailedException;
+import org.apache.kafka.clients.consumer.ConsumerRebalanceListener;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.kafka.configuration.KafkaSourceConfig;
+import org.opensearch.dataprepper.plugins.kafka.configuration.TopicsConfig;
+import org.opensearch.dataprepper.plugins.kafka.source.KafkaSourceBufferAccumulator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Collection;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * * A helper class which helps to process the Plain Text records and write them
+ * into the buffer. Offset handling and consumer re-balancing are also being
+ * handled
+ */
+
+@SuppressWarnings("deprecation")
+public class PlainTextConsumer implements KafkaSourceSchemaConsumer<String, String>, ConsumerRebalanceListener {
+    private static final Logger LOG = LoggerFactory.getLogger(PlainTextConsumer.class);
+    private Map<TopicPartition, OffsetAndMetadata> offsetsToCommit = new HashMap<>();
+    private long lastReadOffset = 0L;
+    private KafkaConsumer<String, String> plainTxtConsumer;
+    private volatile long lastCommitTime = System.currentTimeMillis();
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    @Override
+    public void consumeRecords(final KafkaConsumer<String, String> consumer, final AtomicBoolean status,
+                               final Buffer<Record<Object>> buffer, final TopicsConfig topicConfig,
+                               final KafkaSourceConfig kafkaSourceConfig, PluginMetrics pluginMetrics,
+                               final String schemaType) {
+        KafkaSourceBufferAccumulator kafkaSourceBufferAccumulator = new KafkaSourceBufferAccumulator(topicConfig, kafkaSourceConfig,
+                schemaType, pluginMetrics);
+        plainTxtConsumer = consumer;
+        try {
+            consumer.subscribe(Arrays.asList(topicConfig.getName()));
+            while (!status.get()) {
```

Comment:
this loop is repeated in all consumers. it'll be better if we abstract out in common place and call consumer specific implementation. May be create an Abstract class with common functionality that all consumer will extend.

Reply:
Resolved